src/main/java/dk/cngroup/lentils/controller/HintController.java

```diff
@@ -49,43 +59,61 @@ public String deleteHint(@PathVariable("hintId") final Long hintId) {
         Hint hint = hintService.getHint(hintId);
         Long cypherId = hint.getCypherId();
         hintService.deleteById(hintId);
-        return REDIRECT_HINT_LIST + "?cypherId=" + cypherId;
+        return REDIRECT_HINT_LIST + CYPHERID_PARAMETER + cypherId;
     }
 
     @GetMapping(value = "/update/{hintId}")
-    public String addForm(@PathVariable("hintId") final Long hintId, final Model model) {
-        Hint hint = hintService.getHint(hintId);
-        model.addAttribute(TEMPLATE_ATTR_HEADING, "Upravit hint");
-        model.addAttribute(TEMPLATE_ATTR_HINT, hint);
+    public String updateHintForm(@PathVariable("hintId") final Long hintId, final Model model) {
+        HintFormDTO hintFormDto = mapper.map(hintService.getHint(hintId), HintFormDTO.class);
+        model.addAttribute(TEMPLATE_ATTR_HEADING, HEADING_EDIT_HINT);
+        model.addAttribute(TEMPLATE_ATTR_HINT, hintFormDto);
         return VIEW_HINT;
     }
 
+    @PostMapping(value = "/update/{hintId}")
+    public String updateHint(@PathVariable("hintId") final Long hintId,
+                       @Valid @ModelAttribute(TEMPLATE_ATTR_HINT) final HintFormDTO hintFormDto,
+                       final BindingResult bindingResult,
+                       final Model model) {
+        if (bindingResult.hasErrors()) {
+            model.addAttribute(TEMPLATE_ATTR_HEADING, HEADING_EDIT_HINT);
+            model.addAttribute(TEMPLATE_ATTR_HINT, hintFormDto);
+            return VIEW_HINT;
+        }
+
+        Hint hint = null;
+        try {
+            hint = hintService.getHint(hintId);
+        } catch (ResourceNotFoundException e) {
+            throw e;
+        }
+
+        Hint hintToSave = mapper.map(hintFormDto, Hint.class);
```

Comment:
Tady by bylo dobre, kdyby vsechno to mapovani bylo vykonane v jedne metode. Tj. budes asi muset vytvorit HintMapper, ktery bude overridovat tu metodu map a bude v sobe obsahovat veskere mapovani a nastavovani. tj. misto kodu: 
```java
        Hint hint = null;
        try {
            hint = hintService.getHint(hintId);
        } catch (ResourceNotFoundException e) {
            throw e;
        }

        Hint hintToSave = mapper.map(hintFormDto, Hint.class);
        hintToSave.setHintId(hintId);
        hintToSave.setCypher(hint.getCypher());
        hintService.save(hintToSave);
```
by tam mohlo byt jen:
```java
     hintMapper.map(hintFromDto, Hint.class);
```
a vsechno by se mohlo mapovat uvnitr teto teto metody. Jde o to, ze metoda kontroleru by asi nemela obsahovat pomocne veci(metody, kusy kodu), ktere potrebujes pro to namapovani.

Reply:
@JurcaPavel Tady si to možná trochu překombinoval. Já bych prostě napsal:
```
final Hint hint = hintSevice.getHint(hintId);
mapper.map(hintFormDTO, hint);
hintService.save(hint);
```
Tím pádem odpadá podle mě nutnost mít ten HintMapper. Nebo tady už nevidím fieldy, které by bylo nutné customizovat při tom mapování.

@mjanousek Nebo ne?

Reply:
řekl bych že máš pravdu :D

Reply:
ale vlastně ne protože ta metoda bere hintid a ne hint